src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -7585,6 +7585,8 @@ def where(
                     ]
                 }
         else:
+            frame = frame.ensure_row_position_column()
```

Comment:
Will we need to do this before every join? If so, there is at least one more join in this function, where we don't call `ensure_row_position_column()`. Also, how about `align_on_index()`? I haven't thought about it, so just asking here.

Reply:
More generally, do we need to search for this pattern even outside of `where`?

Reply:
`align_on_index` shouldn't need `row_position` - I also checked to see if anywhere join was being called with a row position column had this same bug (not calling ensure_row_position beforehand) but it doesn't look like this happens anywhere else!

Reply:
Sounds great!